src/main/java/org/prebid/server/bidder/medianet/MedianetBidder.java

```diff
@@ -59,11 +61,28 @@ private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bi
                 .filter(Objects::nonNull)
                 .flatMap(Collection::stream)
                 .filter(Objects::nonNull)
-                .map(bid -> BidderBid.of(bid, resolveBidType(bid.getImpid(), bidRequest.getImp()), currency))
+                .map(bid -> BidderBid.of(bid, resolveBidType(bid, bidRequest.getImp()), currency))
                 .toList();
     }
 
-    private static BidType resolveBidType(String impId, List<Imp> imps) {
+    private static BidType resolveBidType(Bid bid, List<Imp> imps) {
+        final Integer markupType = bid.getMtype();
+        if (markupType == null) {
+            return resolveBidTypeFromImpId(bid.getImpid(), imps);
+        }
+
+        return switch (markupType) {
+            case 1 -> BidType.banner;
+            case 2 -> BidType.video;
+            case 3 -> BidType.audio;
+            case 4 -> BidType.xNative;
+            default ->
+                    throw new PreBidException("Unable to fetch mediaType: %s"
+                            .formatted(bid.getImpid()));
```

Comment:
Please, handle this exception. I don't see these changes on the GO side, so I can't give an exact example, but you can use one of the approaches:
- Handle it per bid, so you can filter out invalid bids.
- Handle it on the response level and fail whole response.

Reply:
Did the changes as suggested. Please check.